Sources/Authenticator/Configuration/AmplifyConfiguration.swift

```diff
@@ -120,6 +120,17 @@ struct CognitoConfiguration {
         case username = "USERNAME"
         case email = "EMAIL"
         case phoneNumber = "PHONE_NUMBER"
+        
+        init?(from authUserAttributeKey: AuthUserAttributeKey) {
+            switch authUserAttributeKey {
+            case .email:
+                self = .email
+            case .phoneNumber:
+                self = .phoneNumber
+            default:
+                return nil
+            }
+        }
     }
```

Comment:
Not especially important right now considering this is scoped to two valid values, but something to think about in the future.

Is there's an opportunity to consolidate to a single source of truth for the raw values with Amplify Auth?
It would likely involve changing some things [here](https://github.com/aws-amplify/amplify-swift/blob/main/AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Support/Utils/AuthUserAttributeKey%2BRawValue.swift).

Reply:
That's indeed an interesting idea to explore. 

Although those raw values are in lowercase and the ones in the JSON configuration are in uppercase, it's still _kinda_ the same.